dubbo-cluster/src/main/java/org/apache/dubbo/rpc/cluster/loadbalance/StatisticsLoadBalance.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.dubbo.rpc.cluster.loadbalance;
+
+import org.apache.dubbo.common.URL;
+import org.apache.dubbo.common.config.ConfigurationUtils;
+import org.apache.dubbo.rpc.cluster.loadbalance.statistics.CpuUsageListener;
+import org.apache.dubbo.rpc.cluster.loadbalance.statistics.CpuUsageService;
+import org.apache.dubbo.rpc.cluster.loadbalance.statistics.CpuUsageServiceImpl;
+import org.apache.dubbo.rpc.Invocation;
+import org.apache.dubbo.rpc.Invoker;
+import org.apache.dubbo.rpc.cluster.loadbalance.statistics.DubboCpuUsageFactory;
+
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+public class StatisticsLoadBalance extends AbstractLoadBalance {
+
+    private Map<String, Float> cpuUsage = new ConcurrentHashMap<>();
+    private DubboCpuUsageFactory dubboCpuUsageFactory;
+    private Map<String, CpuUsageListener> listeners = new ConcurrentHashMap<>();
+
+    public StatisticsLoadBalance() {
+        long timeToLive = Long.parseLong(ConfigurationUtils.getProperty("time.to.live"));
+        long collectCpuUsageInMill = Long.parseLong(ConfigurationUtils.getProperty("mill.to.collect.cpu.usage"));
+        CpuUsageService cpuUsageService = new CpuUsageServiceImpl(timeToLive, collectCpuUsageInMill);
+        cpuUsageService.addListener("statisticsloadbalance", (ip, cpu) -> cpuUsage.put(ip, cpu));
+    }
+
+    @Override
+    protected <T> Invoker<T> doSelect(List<Invoker<T>> invokers, URL url, Invocation invocation) {
+        int index = (int)(invokers.size() * Math.random());
+        Invoker<T> invoker = invokers.get(index);
+        cpuUsage.computeIfAbsent(invoker.getUrl().getAddress(), value -> {
```

Comment:
Listener will be  invalid when provider restart.  An address listener that  removes offline addresses may be needed.

Reply:
Good catch, thanks.